telemetry/definitions/commonDefinitions.json

```
@@ -850,15 +898,38 @@
             "name": "apprunner_viewServiceLogs",
             "description": "View the App Runner service logs (the logs produced by App Runner)"
         },
+        {
+            "name": "auth_addConnection",
+            "description": "Called when user clicks on add New Connections from any of entry points. fields captures snapshot of user state upon reaching a result",
+            "metadata": [
+                { "type": "source"},
+                { "type": "tab", "required": false},
+                { "type": "unlockedTabs", "required": false},
+                { "type": "numberOfConnections", "required": false},
+                { "type": "credentialSourceId", "required": false},
+                { "type": "invalidInputFields", "required": false},
+                { "type": "iamIdentityCenterRegion", "required": false },
+                { "type": "reason", "required": false},
+                { "type": "result"}
+            ]
+        },
+        {
+            "name": "auth_switchBetweenTheRoles",
+            "description": "Called when user clicks yes or no to switch role upon adding new connection",
+            "metadata": [
+                { "type": "numberOfConnections"},
+                { "type": "userChoice"}
```

if it's only "yes or no", would `result` be the right place to store the result? `userChoice` seems like it's intended to store menu selections from many possible choices.

```suggestion
                { "type": "result"}
```

---

- result is predefined to be "success" , "failure" or "cancelled". Since this choice(yes or no) shouldn't be interpreted as success or failure, we will need to add allowedValues for appropriate choices. 
- idea of using user choice is global. we can reuse this anytime we will design metrics for user selection among predefined options.